src/main/java/com/google/devtools/common/options/OptionsParser.java

```diff
@@ -929,4 +934,18 @@ public static boolean getUsesOnlyCoreTypes(Class<? extends OptionsBase> optionsC
     OptionsData data = OptionsParser.getOptionsDataInternal(optionsClass);
     return data.getUsesOnlyCoreTypes(optionsClass);
   }
+
+  /**
+   * A container for an arg and associated options that should be silently ignored when parsed but
+   * not recognized by the current command.
+   */
+  public static final class ArgAndFallbackData {
```

Comment:
Why do we need both this class and the one in `OptionsParserImpl`? Is it something to do with a dependency issue with `OpaqueOptionsData` and `OptionsData`? Seems like we could improve that situation.

Reply:
I thought it was, but by moving the cast into this class I found a way to avoid it and reduce the number of new classes from 3 to 1.

Reply:
Excellent! There's probably still a way we can clean up the class hierarchy, but at least it's not making this change substantially messier now.